# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Hrushi56/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

# 1. My Lane and Why

I selected the **Content Refresh** lane.

Many websites contain thousands of pages, making it difficult for content teams to manually decide which pages should be updated first. The dataset contains information about content age, traffic trends, search performance, and engagement metrics. These features make it possible to identify pages that are likely to benefit from a refresh.

This project focuses on helping teams prioritize their content update efforts instead of relying on manual review.

# 2. The Research Question

## Research Question

Which content pages should be prioritized for refresh to maximize future search performance and user engagement?

## Decision

Decide which pages should be refreshed first.

## Action

The content team updates the highest-priority pages before spending time on lower-impact pages.

## Cost of a Wrong Recommendation

If pages are incorrectly prioritized, the team may spend time updating pages that provide little benefit while important pages continue losing visibility and traffic.

## Why Machine Learning Can Help

With thousands of pages and many performance metrics, manually identifying the best candidates for refresh is difficult. Machine learning can identify patterns across multiple variables to support better prioritization.

In [1]:
import os

print("Current Working Directory:")
print(os.getcwd())

Current Working Directory:
c:\Users\hrutv\Downloads\intershipflyrank-main\flyrank-ml-internship-starter\work\notebooks


## 3. Quick Look at the Data

In [2]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Total Content Pages:", len(df))

print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

print("\nFreshness Tier:")
print(df["freshness_tier"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(), 2))
print("Average Position:", round(df["avg_position"].mean(), 2))

Total Content Pages: 30000

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Freshness Tier:
freshness_tier
0-30      20480
91-180     9171
31-90       175
181+        174
Name: count, dtype: int64

Average CTR: 0.51
Average Position: 16.34


### Interpretation

The dataset contains **30,000 content pages**, which is large enough that manual review is impractical.

More than **16,000 pages are trending down**, suggesting many pages may benefit from updates.

The average CTR is approximately **0.51**, while the average search position is about **16.34**, indicating there is room to improve visibility through better content prioritization.

## 4. Careful Words

This project will recommend which pages should be reviewed for refresh based on historical data.

It does not guarantee that refreshing a page will increase traffic, since search rankings also depend on factors outside this dataset such as competition, search intent, and search engine algorithm updates.

The recommendations should support human decision-making rather than replace it.

## 5. Self Check

- ✅ Selected a provisional lane.
- ✅ Defined the research question.
- ✅ Explained the decision and action.
- ✅ Explained the cost of a wrong recommendation.
- ✅ Included real numbers from the dataset.
- ✅ Explained why machine learning is appropriate.
- ✅ Used careful language about the project's limitations.